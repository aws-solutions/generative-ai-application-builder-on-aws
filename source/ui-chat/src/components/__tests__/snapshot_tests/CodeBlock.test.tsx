/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 **********************************************************************************************************************/

import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { CodeBlock } from '../../CodeBlock';

jest.mock('@cloudscape-design/components', () => {
    const Components = jest.genMockFromModule('@cloudscape-design/components') as any;
    for (const componentName of Object.keys(Components)) {
        Components[componentName] = componentName;
    }
    return Components;
});

const code = `public static void towerOfHanoi(int n, char fromRod,
    char toRod, char auxRod) {
        if (n == 1) {
            System.out.println("Move disk 1 from rod " + 
                fromRod + " to rod " + toRod);
            return;
        }
        towerOfHanoi(n-1, fromRod, auxRod, toRod);
        System.out.println("Move disk " + n + " from rod " +
            fromRod + " to rod " + toRod);
        towerOfHanoi(n-1, auxRod, toRod, fromRod);
    }`;

describe('CodeBlock', () => {
    test('Snapshot test for code block', async () => {
        const tree = renderer.create(<CodeBlock key={Math.random()} language={'java'} value={code} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
