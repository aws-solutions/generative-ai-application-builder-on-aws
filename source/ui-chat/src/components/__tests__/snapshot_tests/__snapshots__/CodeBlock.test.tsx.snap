// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CodeBlock > Snapshot test for code block 1`] = `
<div>
  <div
    className="flex justify-between"
  >
    <span
      className="text-xs lowercase text-white"
    >
      java
    </span>
    <div
      className="flex"
      data-testid="codeblock-button-copy-div"
    >
      <button
        className="flex gap-1.5 items-center rounded text-xs"
        data-testid="codeblock-button-copy"
        onClick={[Function]}
      >
        <svg
          className="tabler-icon tabler-icon-clipboard "
          fill="none"
          height={18}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          viewBox="0 0 24 24"
          width={18}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 5h-2a2 2 0 0 0 -2 2v12a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-12a2 2 0 0 0 -2 -2h-2"
          />
          <path
            d="M9 3m0 2a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v0a2 2 0 0 1 -2 2h-2a2 2 0 0 1 -2 -2z"
          />
        </svg>
        Copy code
      </button>
    </div>
  </div>
  <pre
    style={
      {
        "MozHyphens": "none",
        "MozTabSize": "2",
        "OTabSize": "2",
        "WebkitHyphens": "none",
        "background": "hsl(220, 13%, 18%)",
        "borderRadius": "0.3em",
        "color": "hsl(220, 14%, 71%)",
        "direction": "ltr",
        "fontFamily": ""Fira Code", "Fira Mono", Menlo, Consolas, "DejaVu Sans Mono", monospace",
        "hyphens": "none",
        "lineHeight": "1.5",
        "margin": 0,
        "msHyphens": "none",
        "overflow": "auto",
        "padding": "1em",
        "tabSize": "2",
        "textAlign": "left",
        "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
        "whiteSpace": "pre",
        "wordBreak": "normal",
        "wordSpacing": "normal",
      }
    }
  >
    <code
      className="language-java"
      style={
        {
          "MozHyphens": "none",
          "MozTabSize": "2",
          "OTabSize": "2",
          "WebkitHyphens": "none",
          "background": "hsl(220, 13%, 18%)",
          "color": "hsl(220, 14%, 71%)",
          "direction": "ltr",
          "fontFamily": ""Fira Code", "Fira Mono", Menlo, Consolas, "DejaVu Sans Mono", monospace",
          "hyphens": "none",
          "lineHeight": "1.5",
          "msHyphens": "none",
          "tabSize": "2",
          "textAlign": "left",
          "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
          "whiteSpace": "pre",
          "wordBreak": "normal",
          "wordSpacing": "normal",
        }
      }
    >
      <span
        className="token"
        style={
          {
            "color": "hsl(286, 60%, 67%)",
          }
        }
      >
        public
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(286, 60%, 67%)",
          }
        }
      >
        static
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(286, 60%, 67%)",
          }
        }
      >
        void
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        towerOfHanoi
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        (
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(286, 60%, 67%)",
          }
        }
      >
        int
      </span>
      <span
        style={{}}
      >
         n
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ,
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(286, 60%, 67%)",
          }
        }
      >
        char
      </span>
      <span
        style={{}}
      >
         fromRod
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ,
      </span>
      <span
        style={{}}
      >
        

      </span>
      <span
        style={{}}
      >
            
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(286, 60%, 67%)",
          }
        }
      >
        char
      </span>
      <span
        style={{}}
      >
         toRod
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ,
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(286, 60%, 67%)",
          }
        }
      >
        char
      </span>
      <span
        style={{}}
      >
         auxRod
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        )
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        {
      </span>
      <span
        style={{}}
      >
        

      </span>
      <span
        style={{}}
      >
                
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(286, 60%, 67%)",
          }
        }
      >
        if
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        (
      </span>
      <span
        style={{}}
      >
        n 
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        ==
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(29, 54%, 61%)",
          }
        }
      >
        1
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        )
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        {
      </span>
      <span
        style={{}}
      >
        

      </span>
      <span
        style={{}}
      >
                    
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(29, 54%, 61%)",
          }
        }
      >
        System
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        .
      </span>
      <span
        style={{}}
      >
        out
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        .
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        println
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        (
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(95, 38%, 62%)",
          }
        }
      >
        "Move disk 1 from rod "
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        +
      </span>
      <span
        style={{}}
      >
         

      </span>
      <span
        style={{}}
      >
                        fromRod 
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        +
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(95, 38%, 62%)",
          }
        }
      >
        " to rod "
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        +
      </span>
      <span
        style={{}}
      >
         toRod
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        )
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ;
      </span>
      <span
        style={{}}
      >
        

      </span>
      <span
        style={{}}
      >
                    
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(286, 60%, 67%)",
          }
        }
      >
        return
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ;
      </span>
      <span
        style={{}}
      >
        

      </span>
      <span
        style={{}}
      >
                
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        }
      </span>
      <span
        style={{}}
      >
        

      </span>
      <span
        style={{}}
      >
                
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        towerOfHanoi
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        (
      </span>
      <span
        style={{}}
      >
        n
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        -
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(29, 54%, 61%)",
          }
        }
      >
        1
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ,
      </span>
      <span
        style={{}}
      >
         fromRod
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ,
      </span>
      <span
        style={{}}
      >
         auxRod
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ,
      </span>
      <span
        style={{}}
      >
         toRod
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        )
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ;
      </span>
      <span
        style={{}}
      >
        

      </span>
      <span
        style={{}}
      >
                
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(29, 54%, 61%)",
          }
        }
      >
        System
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        .
      </span>
      <span
        style={{}}
      >
        out
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        .
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        println
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        (
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(95, 38%, 62%)",
          }
        }
      >
        "Move disk "
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        +
      </span>
      <span
        style={{}}
      >
         n 
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        +
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(95, 38%, 62%)",
          }
        }
      >
        " from rod "
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        +
      </span>
      <span
        style={{}}
      >
        

      </span>
      <span
        style={{}}
      >
                    fromRod 
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        +
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(95, 38%, 62%)",
          }
        }
      >
        " to rod "
      </span>
      <span
        style={{}}
      >
         
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        +
      </span>
      <span
        style={{}}
      >
         toRod
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        )
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ;
      </span>
      <span
        style={{}}
      >
        

      </span>
      <span
        style={{}}
      >
                
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        towerOfHanoi
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        (
      </span>
      <span
        style={{}}
      >
        n
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(207, 82%, 66%)",
          }
        }
      >
        -
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(29, 54%, 61%)",
          }
        }
      >
        1
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ,
      </span>
      <span
        style={{}}
      >
         auxRod
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ,
      </span>
      <span
        style={{}}
      >
         toRod
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ,
      </span>
      <span
        style={{}}
      >
         fromRod
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        )
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        ;
      </span>
      <span
        style={{}}
      >
        

      </span>
      <span
        style={{}}
      >
            
      </span>
      <span
        className="token"
        style={
          {
            "color": "hsl(220, 14%, 71%)",
          }
        }
      >
        }
      </span>
    </code>
  </pre>
</div>
`;
